### DISCUSSION TITLE
Options API - mapState property as function - type issue

### DISCUSSION THREAD

**Discussion by memmm** [vuejs/pinia#588] [Help and Questions] [answered]
I'm using the Options API and getting the store state with mapState. I would like to use a function to access the values as described [here](https://pinia.esm.dev/core-concepts/state.html#usage-with-the-options-api). The property's type is correct in mapState:

![typing](https://user-images.githubusercontent.com/21298183/127334013-957d92d3-ae00-431a-ab48-f9443a4331c2.png)

When accessing the value in a method however, I get ```Type 'boolean' is not assignable to type 'never'.``` With string-style access, there is no issue.

![typing2](https://user-images.githubusercontent.com/21298183/127334029-dd8fcb76-97ef-41a3-895d-b1f3de329a87.png)

Why is it cast as type never?

**Comment by posva** [comment]
What version of pinia is this? Tests are currently passing

**Reply by memmm** [reply]
I'm using 2.0.0-beta.5 (next)

**Reply by posva** [reply]
It should be fixed in the next version then (which requires vue 3.2.0 currently in beta). The only other possibility I see is having `getters` incorrectly defined

**Comment by posva** [comment, marked as answer]
Should be fixed on the latest rc release
